source/includes/library-reference/_brain.html.md

```diff
@@ -0,0 +1,47 @@
+# Brain Class
+
+Used to manage a BRAIN instance and talk with the server backend. Contains
+state information about the BRAIN and can be used to upload/download inkling
+to and from the BRAIN on the server.
+
+Requires a shared pointer to a `bonsai::Config` and a BRAIN name.
+
+## Brain(config, name)
+
+Construct a BRAIN using an existing configuration and BRAIN name.
+
+- `config` Shared pointer to previously created `bonsai::Config`.
+- `name`   BRAIN name as specified on the server. If name is empty the
+            BRAIN name in `config` will be used instead.
+
+## setup()
+
+Checks the `config()` and starts up a BRAIN for training if running in training mode.
+Stops a BRAIN running for training mode if requested for prediction and the version is the latest.
+
+## update()
+
+Retrieves current state information about the BRAIN from the server
+
+## bool ready()
```

Comment:
Python doesn't usually document return types. See SQL alchemy for example python docs: http://docs.sqlalchemy.org/en/latest/core/connections.html#sqlalchemy.engine.Engine
Python doc styleguide is here but relies on restructuredText/sphinx. https://devguide.python.org/documenting/

Reply:
not python docs.